R/utils.R

```diff
@@ -1,3 +1,314 @@
+#' Assign an additional name to an entity
+#'
+#' * `to_alias()` takes any character vector and creates an alias by
+#' transforming the input (a) to lower case; (b) to latin-ascii characters; and
+#' (c) to standard abbreviations of ownership types. Commonly, the inputs are
+#' values from the columns `name_direct_loantaker` or `name_ultimate_parent`
+#' of a loanbook dataset, or from the column `name_company` of an asset-level
+#' dataset.
+#' * `from_name_to_alias()` outputs a table giving default strings used to
+#' convert from a name to its alias. You may amend this table and pass it to
+#' `to_alias()` via the `from_to` argument.
+#'
+#' @section Assigning aliases:
+#' The transformation process used to compare names between loanbook and tilt
+#' datasets applies best practices commonly used in name matching algorithms:
+#' * Remove special characters.
+#' * Replace language specific characters.
+#' * Abbreviate certain names to reduce their importance in the matching.
+#' * Spell out numbers to increase their importance.
+#'
+#' @author person(given = "Evgeny", family = "Petrovsky", role = c("aut",
+#'   "ctr"))
+#'
+#' @param x Character string, commonly from the columns `name_direct_loantaker`
+#'   or `name_ultimate_parent` of a loanbook dataset, or from the column
+#'   `name_company` of an asset-level dataset.
+#' @param from_to A data frame with replacement rules to be applied, contains
+#'   columns `from` (for initial values) and `to` (for resulting values).
+#' @param ownership vector of company ownership types to be distinguished for
+#'   cut-off or separation.
+#' @param remove_ownership Flag that defines whether ownership type (like llc)
+#'   should be cut-off.
+#'
+#' @return
+#' * `to_alias()` returns a character string.
+#' * `from_name_to_alias()` returns a [tibble::tibble] with columns `from` and
+#' `to`.
+#'
+#' @examples
+#' library(dplyr)
+#'
+#' to_alias("A. and B")
+#' to_alias("Acuity Brands Inc")
+#' to_alias(c("3M Company", "Abbott Laboratories", "AbbVie Inc."))
+#'
+#' custom_replacement <- tibble(from = "AAAA", to = "B")
+#' to_alias("Aa Aaaa", from_to = custom_replacement)
+#'
+#' neutral_replacement <- tibble(from = character(0), to = character(0))
+#' to_alias("Company Name Owner", from_to = neutral_replacement)
+#' to_alias(
+#'   "Company Name Owner",
+#'   from_to = neutral_replacement,
+#'   ownership = "owner",
+#'   remove_ownership = TRUE
+#' )
+#'
+#' from_name_to_alias()
+#'
+#' append_replacements <- from_name_to_alias() %>%
+#'   add_row(
+#'     .before = 1,
+#'     from = c("AA", "BB"), to = c("alpha", "beta")
+#'   )
+#' append_replacements
+#'
+#' # And in combination with `to_alias()`
+#' to_alias(c("AA", "BB", "1"), from_to = append_replacements)
+#' @export
+to_alias <- function(x,
```

Comment:
Since we copy larger parts from r2di..match I would suggest to give some credit, at least by mentioning it in the description but @maurolepore might know a more standard way to do this?

Reply:
More tan credit I would focus on source. The licence is very flexible and you can do just about anything with that code. But I think it's useful to write a comment pointing to the source as precisely as possible, or example:

```
# Source: https://github.com/RMI-PACTA/r2dii.match/blob/8e2b8a051145d74ae34f1d8bf35a8bd0b71a3ce8/R/to_alias.R
```

If you want to include it in the exported documentation you may use `#' @source` and maybe refer to the latest release on CRAN, for example:

```r
#' @source [r2dii.match](https://cran.r-project.org/package=r2dii.match) version 0.1.3.
```